hs_tools_resource/forms.py

```diff
@@ -134,21 +134,73 @@ def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, *
 
-                    self.fields['supported_res_types'].initial = checked_item_list
+                    self.initial['supported_res_types'] = \
+                        [parameter.description for parameter in supported_res_types]
                 else:
-                    self.fields['supported_res_types'].initial = []
+                    self.initial['supported_res_types'] = []
             except:
-                self.fields['supported_res_types'].initial = []
+                self.initial['supported_res_types'] = []
 
     class Meta:
         model = SupportedResTypes
-        fields = '__all__'
+        fields = ('supported_res_types',)
 
 
 class SupportedResTypesValidationForm(forms.Form):
-    supported_res_types = forms.MultipleChoiceField(choices=parameters_choices, required=False)
+    supported_res_types = forms.MultipleChoiceField(choices=SupportedResTypes_choices, required=False)
 
+
+SupportedSharingStatus_choices = (
+    ('Published', 'Published'),
+    ('Public', 'Public'),
+    ('Discoverable', 'Discoverable'),
+    ('Private', 'Private'),
+)
+
+
+class SupportedSharingStatusFormHelper(BaseFormHelper):
+    def __init__(self, allow_edit=True, res_short_id=None,
+                 element_id=None, element_name=None,  *args, **kwargs):
+
+        # the order in which the model fields are listed for
+        # the FieldSet is the order these fields will be displayed
+        layout = Layout(MetadataField('sharing_status'))
+        kwargs['element_name_label'] = 'Supported Resource Sharing Status'
+        super(SupportedSharingStatusFormHelper, self).\
+            __init__(allow_edit, res_short_id, element_id,
+                     element_name, layout,  *args, **kwargs)
+
+
+class SupportedSharingStatusForm(ModelForm):
+    sharing_status = forms.MultipleChoiceField(choices=SupportedSharingStatus_choices,
+                                               widget=forms.CheckboxSelectMultiple(
+                                                attrs={'style': 'width:auto;margin-top:-5px'}))
+
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(SupportedSharingStatusForm, self).__init__(*args, **kwargs)
+        self.fields['sharing_status'].label = "Choose Sharing Status:"
+        self.helper = SupportedSharingStatusFormHelper(allow_edit, res_short_id, element_id,
+                                                       element_name='SupportedSharingStatus')
+        if self.instance:
+            try:
+                supported_sharing_status = self.instance.sharing_status.all()
+                if len(supported_sharing_status) > 0:
+                    self.initial['sharing_status'] = \
+                        [parameter.description for parameter in supported_sharing_status]
+                else:
+                    self.initial['sharing_status'] = []
+            except:
+                self.initial['sharing_status'] = []
```

Comment:
@pkdash I just found a issue with webapp on beta, and looks like this try...catch statement is needed here. If a new webapp is just created, after switching to edit mode, a ValueError exception will be raised here.